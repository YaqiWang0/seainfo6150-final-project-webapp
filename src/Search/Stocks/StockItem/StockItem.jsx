import React from "react"
import style from "./StockItem.module.css"
import { useHistory, useLocation } from "react-router-dom";

const StockItem = (props) => {
    const randomClass = () => {
        if(props.test === true)
            return "container1"
        let name = "container"
        name = name + `${Math.floor((Math.random() * 6) + 1)}`;
        return name
    }

    let history = useHistory();
    let location = useLocation();

    const openDetail = async (e) => {
        const response = await fetch(`https://financialmodelingprep.com/api/v3/profile/${props.stock.symbol}?apikey=56f622a93b55019c5a0875058fc58f13`);
        const responseJson = await response.json();
        const newsResponse = await fetch(`https://financialmodelingprep.com/api/v3/stock_news?tickers=${props.stock.symbol}&limit=10&apikey=56f622a93b55019c5a0875058fc58f13`);
        const newsResponseJson = await newsResponse.json();
        history.push(
            "/company",
            {
                info: Object.values(responseJson),
                news: newsResponseJson,
                login: props.login
            }
        )
    }
    return (
        <div className={style[randomClass()]}>
                <div className={style.name} onClick={openDetail}>
                    <span className={style.stock_name}>{props.stock.symbol}</span>
                    <span className={style.company_name}>{props.stock.name}</span>
                </div>
        </div>
    )
}

export default StockItem