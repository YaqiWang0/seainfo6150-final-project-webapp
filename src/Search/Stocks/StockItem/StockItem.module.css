.container1 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 1em;
    width: 100%;

}

.container2 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 30%;
    width: 100%;
}

.container3 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 50%;
    width: 100%;
}

.container4 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 70%;
    width: 100%;
}

.container5 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 80%;
    width: 100%;
}

.container6 {
    display: flex;
    flex-direction: row;
    border-radius: 5%;
    padding: 1em 1em 1em 20%;
    width: 100%;
}

.name {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    opacity: 0.5;

    transition: transform .5s ease;
    animation: fadein 2s;

    -webkit-animation: fadein 2s;
    cursor: pointer;
}

.name:hover {
    transform: scale(1.5);
    opacity: 1;
}

.stock_name {
    font-size: 4vmin;
    color: white;
    text-align: left;
    width: 100%;
    cursor: inherit;
}

.company_name {
    color: gray;
    font-family: Serif;
    width: 100%;
    display: block;
    overflow: hidden;
    white-space: nowrap;
    text-align: left;
    font-size: 1.5vmin;
    cursor: inherit;
}

@keyframes fadein {
    from {
        opacity:0;
    }
    to {
        opacity:1;
    }
}
@-moz-keyframes fadein { /* Firefox */
    from {
        opacity:0;
    }
    to {
        opacity:1;
    }
}
@-webkit-keyframes fadein { /* Safari and Chrome */
    from {
        opacity:0;
    }
    to {
        opacity:1;
    }
}
@-o-keyframes fadein { /* Opera */
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}