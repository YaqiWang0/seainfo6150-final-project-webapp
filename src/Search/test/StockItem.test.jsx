import React from "react";
import data from "./singleData";
import StockItem from "../Stocks/StockItem/StockItem";

describe("StockItem test", () => {

    it('render successfully', function () {
        const {container} = render(
            <StockItem stock = {data} test = {true}/>
        );
        expect(container).toMatchSnapshot();
    })
})