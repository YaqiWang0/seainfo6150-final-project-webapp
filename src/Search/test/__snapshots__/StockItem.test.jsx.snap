// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockItem test render successfully 1`] = `
<div>
  <div
    class="container1"
  >
    <div
      class="name"
    >
      <span
        class="stock_name"
      >
        TSLA
      </span>
      <span
        class="company_name"
      >
        Tesla, Inc.
      </span>
    </div>
  </div>
</div>
`;
